bundles/org.openhab.binding.ondilo/src/main/java/org/openhab/binding/ondilo/internal/OndiloApiClient.java

```diff
@@ -0,0 +1,103 @@
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.core.auth.client.oauth2.AccessTokenResponse;
+import org.openhab.core.auth.client.oauth2.OAuthClientService;
+import org.openhab.core.auth.client.oauth2.OAuthException;
+import org.openhab.core.auth.client.oauth2.OAuthResponseException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link OndiloApiClient} for accessing the Ondilo API using OAuth2 authentication.
+ * handlers.
+ *
+ * @author MikeTheTux - Initial contribution
+ */
+@NonNullByDefault
+public class OndiloApiClient {
+    private final Logger logger = LoggerFactory.getLogger(OndiloApiClient.class);
+    private @Nullable OAuthClientService oAuthService;
+    private @Nullable String bearer;
+    private @Nullable AccessTokenResponse accessTokenResponse;
+    private static final String ONDILO_API_URL = "https://interop.ondilo.com/api/customer/v1";
+
+    public OndiloApiClient(OAuthClientService oAuthService, AccessTokenResponse accessTokenResponse) {
+        this.oAuthService = oAuthService;
+        this.accessTokenResponse = accessTokenResponse;
+        this.bearer = accessTokenResponse.getAccessToken();
+        logger.trace("OndiloApiClient initialized with OAuth2 service and bearer token");
+    }
+
+    @Nullable
+    public synchronized String get(String endpoint) {
+        try {
+            refreshAccessTokenIfNeeded();
+            URL url = new URI(ONDILO_API_URL + endpoint).toURL();
+            HttpURLConnection conn = (HttpURLConnection) url.openConnection();
+            conn.setRequestMethod("GET");
+            conn.setRequestProperty("Authorization", "Bearer " + bearer);
+            conn.setRequestProperty("Accept", "application/json");
+            conn.setRequestProperty("Accept-Charset", "utf-8");
+            conn.setRequestProperty("Accept-Encoding", "gzip, deflate");
+            conn.connect();
+            int responseCode = conn.getResponseCode();
+            if (responseCode == 200) {
+                try (InputStream is = conn.getInputStream(); Scanner scanner = new Scanner(is, "UTF-8")) {
+                    String response = scanner.useDelimiter("\\A").next();
+                    return response;
+                }
+            } else {
+                logger.error("Ondilo API request failed with code: {}", responseCode);
+            }
+        } catch (InterruptedIOException e) {
+            logger.warn("Ondilo API request interrupted: {}", e.getMessage());
+            Thread.currentThread().interrupt();
+        } catch (IOException | URISyntaxException e) {
+            logger.error("Ondilo API request error", e);
+        }
+        return null;
+    }
+
+    private void refreshAccessTokenIfNeeded() {
+        OAuthClientService oAuthService = this.oAuthService;
+        AccessTokenResponse accessTokenResponse = this.accessTokenResponse;
+        if ((oAuthService != null) && (accessTokenResponse != null)) {
```

Comment:
```suggestion
        if (oAuthService != null && accessTokenResponse != null) {
```

Reply:
This is my coding guideline that I'm using all over the binding ...

Reply:
And i don't know why, as it is not needed at all. Please adapt.

Reply:
done